### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by vermasavita**
### What do you need to check?

- Login functionality
- Signup functionality
- Logout functionality
- Require Auth

Note: Please ignore login & signup page UI. I'll fix that later

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1 @@
+REACT_APP_JWT_SECRET=reactapp
```
Comment (manish591) [inline .env:1]: don't add env in commits.

File: src/App.js [hunk 2]
```diff
@@ -1,19 +1,43 @@
 import { BrowserRouter as Router, Routes, Route } from "react-router-dom";
 import { Home, Login, Signup, Bookmark, AllPost, Profile } from "./pages";
+import { ToastContainer } from "react-toastify";
+import { RequireAuth } from "./RequiresAuth";
+import "react-toastify/dist/ReactToastify.css";
 function App() {
```
Comment (manish591) [inline src/App.js]: You are using function declaration in your app component, While in the rest of the component you are using the arrow function syntax. So you can change this to arrow function as well to maintain consistency.

File: src/pages/authentication/Login.jsx [hunk 3]
```diff
@@ -1,7 +1,41 @@
 import logo from "../../Assets/Friends.png";
 import { NavLink } from "react-router-dom";
 import "./Auth.css";
+import { useState, useEffect } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { loginUser } from "./authSlice";
+import { useNavigate } from "react-router-dom";
+import { toast } from "react-toastify";
 const Login = () => {
```
Comment (manish591) [inline src/pages/authentication/Login.jsx:10]: add some whitespace between import statements and component

**Review by manish591 — APPROVED** [review]
The code is looking good. 
For consistent formatting use prettier.

**Review by ravikumar1002 — APPROVED** [review]
looks good to merge

File: .env (hunk 1 above)
Comment (vermasavita) [reply .env:1]: Noted

File: src/App.js (hunk 2 above)
Comment (vermasavita) [reply src/App.js]: Changed. Thanks :-)

File: src/pages/authentication/Login.jsx (hunk 3 above)
Comment (vermasavita) [reply src/pages/authentication/Login.jsx:10]: Done

File: src/App.js [hunk 4]
```diff
@@ -1,19 +1,44 @@
...
-                <AllPost />
-              </Home>
+              <RequireAuth>
+                <Home>
+                  <AllPost />
+                </Home>
+              </RequireAuth>
             }
           />
           <Route
```
Comment (theVivek-kumar) [inline src/App.js:44]: use outlet for routes .< element is used as a placeholder. In this case an <Outlet> enables the Users component to render its child routes>

**Review by theVivek-kumar — APPROVED** [review]

File: src/App.js (hunk 4 above)
Comment (vermasavita) [reply src/App.js:44]: I did not get you. Can you please explain again?

**Pull request merged**